lib/hunter/commands/hunt.rb

```diff
@@ -75,7 +75,8 @@ def run
             port: @port,
             server: Config.target_host || 'localhost',
             auth: @auth_key,
-            broadcast: false
+            broadcast: false,
+            groups: []
```

Comment:
Why do we need to set this to an empty array instead of just omitting it?

Reply:
It seems that, when a command line option is an array, leaving it without any values will cause it to default to an empty array instead of just `nil`.

Initially, the preset data in `send` was determined like so:
```
label: @options.label || Config.presets["label"],
prefix: @options.prefix || Config.presets["prefix"],
groups: @options.groups || Config.presets["groups"],
```
however, this ended up creating inconsistent behaviour when `--groups` was not used, with `--include-self` using the config value for `groups` and `send` using an empty array every time. Given that `--include-self` is emulating a normal instance of `send`, it makes sense for `--include-self`'s implementation to change, instead of having `send` change its `groups` option value to `nil` when none are given.

Reply:
> It seems that, when a command line option is an array, leaving it without any values will cause it to default to an empty array instead of just nil.

Before I get to the rest of this, why are we setting it to be an array to begin with? It's impossible for a non-boolean CLI option to be anything but a string. The expected input for `send --groups` is a comma-separated list of group names that we should be splitting internally.

Reply:
Scratch that comment, I didn't know we were using `c.slop.array` :upside_down_face: